flask_resources/serializers/json.py

```diff
@@ -37,7 +37,7 @@ def default(self, obj):
         """Override parent's default."""
         if is_lazy_string(obj):
             return str(obj)
-        return super().default(obj)
+        return _default(obj)
```

Comment:
Note for reviewers:

The `flask.json.provider._default` implementation throws an exception if none of the conditions is met ([ref](https://github.com/pallets/flask/blob/main/src/flask/json/provider.py#L120)). However, the old implementation of JSONEncoder did a `super.default()` ([ref](https://github.com/pallets/flask/blob/a74e266474f0981e9fc0d54d5dc430386823e4d7/src/flask/json/__init__.py#L50)). This default would throw the `TypeError` thrown by `_default` ([ref](https://docs.python.org/3/library/json.html#json.JSONEncoder.default)). Assuming that iterables are serialized correctly in the _dataclasses_ condition, then the behaviour is analog.